collections/nemo_nlp/nemo_nlp/data/datasets/bert_pretraining.py

```diff
@@ -216,81 +203,71 @@ def match_target_seq_length(document, target_seq_length, filename,
-            b_document, b_line_idx = match_target_seq_length(
-                b_document, target_seq_length_b, b_filename, b_line_idx,
-                self.sentence_indices)
 
-        def truncate_seq_pair(a_document, b_document, max_num_tokens):
-            # Truncates a pair of sequences to a maximum sequence length
-            while True:
-                total_length = len(a_document) + len(b_document)
-                if total_length <= max_num_tokens:
-                    break
+        is_next = int(not take_random_b)
+        b_line_pos = self.sentence_indices[b_filename][b_line_idx]
+        b_document = get_document(b_filename, b_line_pos)
+        b_document, b_line_idx = match_target_seq_length(
+            b_document, target_seq_length_b, b_filename, b_line_idx,
+            self.sentence_indices)
 
-                trunc_document = a_document if len(
-                    a_document) > len(b_document) else b_document
-                if len(trunc_document) <= 1:
-                    raise ValueError("Input text corpora probably too small. "
-                                     "Failed to truncate sequence pair to "
-                                     "maximum sequence length.")
+        def truncate_seq_pair(a, b, max_num_tokens):
+            # Truncates a pair of sequences to a maximum sequence length
+            while (len(a) + len(b)) > max_num_tokens:
+                # Truncate the longer sequence
+                if len(a) > len(b):
+                    trunc_document = a
+                else:
+                    trunc_document = b
 
+                # Randomly truncate from the front or the back
                 if random.random() < 0.5:
                     del trunc_document[0]
                 else:
                     trunc_document.pop()
 
+                total_length = len(a_document) + len(b_document)
+
         truncate_seq_pair(a_document, b_document, max_num_tokens)
 
-        output_ids = [self.tokenizer.special_tokens["[CLS]"]] + a_document + \
-                     [self.tokenizer.special_tokens["[SEP]"]] + b_document + \
-                     [self.tokenizer.special_tokens["[SEP]"]]
+        output_ids = [self.cls_id] + a_document + \
+                     [self.sep_id] + b_document + [self.sep_id]
 
         input_ids, output_mask = self.mask_ids(output_ids)
 
-        output_mask = np.array(output_mask, dtype=np.float32)
         input_mask = np.zeros(self.max_seq_length, dtype=np.float32)
         input_mask[:len(input_ids)] = 1
 
         input_type_ids = np.zeros(self.max_seq_length, dtype=np.int)
         input_type_ids[len(a_document) + 2:len(output_ids) + 1] = 1
 
         while len(input_ids) < self.max_seq_length:
-            input_ids.append(0)
-            output_ids.append(0)
-            output_mask = np.append(output_mask, [0])
+            input_ids.append(self.pad_id)
+            output_ids.append(self.pad_id)
+            output_mask.append(0)
```

Comment:
extra = max(0, self.max_seq_length - input_ids)
inputs_ids.extend([self.pad_id] * extra)
output_ids.extend([self.pad_id] * extra)
output_mask.extend([0] * extra)

Reply:
yes. improved as you suggested.